packages/app-store/zohocalendar/api/callback.ts

```diff
@@ -0,0 +1,82 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { stringify } from "querystring";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { getSafeRedirectUrl } from "@calcom/lib/getSafeRedirectUrl";
+import logger from "@calcom/lib/logger";
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+import { decodeOAuthState } from "../../_utils/decodeOAuthState";
+import getAppKeysFromSlug from "../../_utils/getAppKeysFromSlug";
+import getInstalledAppPath from "../../_utils/getInstalledAppPath";
+import config from "../config.json";
+import type { ZohoAuthCredentials } from "../types/ZohoCalendar";
+import { appKeysSchema as zohoKeysSchema } from "../zod";
+
+const log = logger.getChildLogger({ prefix: [`[[zohocalendar/api/callback]`] });
+
+async function getHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { code } = req.query;
+  const state = decodeOAuthState(req);
+
+  if (code && typeof code !== "string") {
+    res.status(400).json({ message: "`code` must be a string" });
+    return;
+  }
+
+  if (!req.session?.user?.id) {
+    return res.status(401).json({ message: "You must be logged in to do this" });
+  }
+
+  const appKeys = await getAppKeysFromSlug(config.slug);
+  const { client_id, client_secret } = zohoKeysSchema.parse(appKeys);
+
+  const params = {
+    client_id,
+    grant_type: "authorization_code",
+    client_secret,
+    redirect_uri: `${WEBAPP_URL}/api/integrations/${config.slug}/callback`,
+    code,
+  };
+
+  const query = stringify(params);
+
+  const response = await fetch(`https://accounts.zoho.com/oauth/v2/token?${query}`, {
```

Comment:
we can have a variable for the base url here `https://accounts.zoho.com/oauth/v2` so that it can be updated later
this is referenced in `/api/add` and `/api/callback` both

Reply:
done